board/raspberrypi3/overlay/usr/bin/streameye.sh

```diff
@@ -11,64 +12,272 @@ test -r ${RASPIMJPEG_CONF} || exit 1
+    elif [ "${awb}" = "auto" ]; then awb="1"
+    elif [ "${awb}" = "sunlight" ]; then awb="6"
+    elif [ "${awb}" = "cloudy" ]; then awb="8"
+    elif [ "${awb}" = "shade" ]; then awb="9"
+    elif [ "${awb}" = "tungsten" ]; then awb="4"
+    elif [ "${awb}" = "fluorescent" ]; then awb="3"
+    elif [ "${awb}" = "incandescent" ]; then awb="2"
+    elif [ "${awb}" = "flash" ]; then awb="7"
+    elif [ "${awb}" = "horizon" ]; then awb="5"
+    else awb="1"
+    fi
+    v4l2-ctl ${video_arg} --set-ctrl=white_balance_auto_preset=${awb} &>/dev/null
+
+    metering=$(grep -e ^metering ${RASPIMJPEG_CONF} | cut -d ' ' -f 2)
+    if [ "${metering}" = "spot" ]; then metering="2"
+    elif [ "${metering}" = "center" ]; then metering="1"
+    else metering="0"
+    fi
+    v4l2-ctl ${video_arg} --set-ctrl=exposure_metering_mode=${awb} &>/dev/null
+
+    scene="0"
+    exposure=$(grep -e ^exposure ${RASPIMJPEG_CONF} | cut -d ' ' -f 2)
+    if [ "${exposure}" = "off" ]; then
+        exposure="1"
+    else
+        if [ "${exposure}" = "night" ]; then scene="8"
+        elif [ "${exposure}" = "backlight" ]; then scene="1"
+        elif [ "${exposure}" = "sports" ]; then scene="11"
+        elif [ "${exposure}" = "snow" ]; then scene="2"
+        elif [ "${exposure}" = "beach" ]; then scene="2"
+        elif [ "${exposure}" = "fireworks" ]; then scene="6"
         fi
-        raspimjpeg_opts="${raspimjpeg_opts} --${line}"
-    done < ${RASPIMJPEG_CONF}
+        exposure="0"
+    fi
+    v4l2-ctl ${video_arg} --set-ctrl=auto_exposure=${exposure} &>/dev/null
+    v4l2-ctl ${video_arg} --set-ctrl=scene_mode=${scene} &>/dev/null
 
+    ev=$(grep -e ^ev ${RASPIMJPEG_CONF} | cut -d ' ' -f 2)
+    if [ -z "${ev}" ]; then ev="0"; fi
+    ev=$(expr ${ev} + 25)
+    ev=$(expr ${ev} / 2)
+    if [ "${ev}" -gt 24 ]; then ev="24"; fi
+    v4l2-ctl ${video_arg} --set-ctrl=auto_exposure_bias=${ev} &>/dev/null
+
+    video_bitrate=$(grep -e ^bitrate ${RASPIMJPEG_CONF} | cut -d ' ' -f 2)
+    if [ -z "${video_bitrate}" ]; then video_bitrate="1000000"; fi
+    v4l2-ctl ${video_arg} --set-ctrl=video_bitrate=${video_bitrate} &>/dev/null
+}
+
+invalid_opt () {
+    local e match="$1"
+    shift
+    for e; do [[ "${e}" == "${match}" ]] && return 1; done
+    return 0
+}
+
+function start() {
     source ${STREAMEYE_CONF}
     streameye_opts="-p ${PORT}"
     if [ -n "${CREDENTIALS}" ] && [ "${AUTH}" = "basic" ]; then
         streameye_opts="${streameye_opts} -a basic -c ${CREDENTIALS}"
     fi
 
-    if [ -r ${MOTIONEYE_CONF} ] && grep 'log-level debug' ${MOTIONEYE_CONF} >/dev/null; then
-        raspimjpeg_opts="${raspimjpeg_opts} -d"
-        streameye_opts="${streameye_opts} -d"
-    fi
+    if [ "${PROTO}" = "rtsp" ]; then
+        pid=$(ps | grep test-launch | grep -v grep | tr -s ' ' | sed -e 's/^\s//' | cut -d ' ' -f 1)
+        if [ -n "${pid}" ]; then
+            return
+        fi
+
+        iptables -A INPUT -p tcp -s localhost --dport 554 -j ACCEPT
```

Comment:
@jasaw why do we need to adjust the firewall settings? By default, port 554 is open and, if user toys with the firewall, they could break many other things (such as ports 80 or 8081), for which motionEyeOS does not have any kind of protection.

Reply:
@ccrisan see known issue 1. However, when I wrote that comment, I thought it was the RTSP server issue. Since then, I realized that my wifi dongle has been playing up, so now I'm not sure issue 1 is caused by my wifi dongle or RTSP server. Need more testing.

Reply:
I'm fine with the iptables workaround - I don't see any harm in it, given that the only port affected is `554`.